### PR TITLE
feat!: bring dev server headers to parity with production

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

Some of the headers returned by the CLI's dev server are different from what we return in production. This PR attempts to bring them closer.

### CODE REVIEW

File: src/lib/edge-functions/proxy.mjs [hunk 1]
```diff
@@ -133,7 +132,6 @@ export const initializeProxy = async ({
       [headers.Functions]: functionNames.join(','),
       [headers.ForwardedHost]: `localhost:${passthroughPort}`,
       [headers.Passthrough]: 'passthrough',
-      [headers.RequestID]: generateUUID(),
```
Comment (eduardoboucas) [inline src/lib/edge-functions/proxy.mjs:136]: Moved to the proxy so that all requests have it.

**Review by khendrikse — DISMISSED** [review]

**Comment by eduardoboucas** [comment]
@khendrikse I added and updated some tests so your review got stale. Could you give it another look, please?

**Review by khendrikse — APPROVED** [review]

File: src/utils/request-id.mjs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// @ts-check
+import { v4 as generateUUID } from 'uuid'
+
+// Transform a v4 UUID to match the format used in production — alphanumeric
+// characters, all uppercase, 26 characters.
+export const generateRequestID = () => generateUUID().replace(/-/g, '').toUpperCase().slice(0, 26)
```
Comment (Skn0tt) [inline src/utils/request-id.mjs:6]: if we wanted, we *could* use https://github.com/ulid/javascript to get the very same format as is being used in production, so if any of our users rely on the characteristics of ULIDs (e.g. the embedded timestamp), that'd make it easier to test in dev. I don't think anybody does that, nor that anybody *should*, so it's probably right to use UUID here. And it sames some NPM dead weight.

File: src/utils/static-server.mjs [hunk 3]
```diff
@@ -21,7 +21,8 @@ export const startStaticServer = async ({ settings }) => {
   })
 
   server.addHook('onRequest', (req, reply, done) => {
-    reply.header('X-Powered-by', 'netlify-dev')
```
Comment (Skn0tt) [inline src/utils/static-server.mjs:24]: should we add the `Server: Netlify` header? or `Server: Netlify Dev`, as we call it in https://docs.netlify.com/cli/get-started/#get-started-with-netlify-dev?

**Review by Skn0tt — DISMISSED** [review]
looks good, left two minor non-blocking comments.

File: src/utils/request-id.mjs (hunk 2 above)
Comment (eduardoboucas) [reply src/utils/request-id.mjs]: Yeah I was torn on this myself. But if the idea is to bring parity with production as closely as possible, we might as well use the same format. Updated in https://github.com/netlify/cli/pull/5490/commits/d0227f71455c0aa8291cd3717760ebb590fc510f.

File: src/utils/static-server.mjs (hunk 3 above)
Comment (eduardoboucas) [reply src/utils/static-server.mjs:24]: Unless there's a reason to differentiate them, I'd rather use the same value in both environments.

**Comment by eduardoboucas** [comment]
Changed to `feat!` as this is technically a breaking change.

File: src/utils/static-server.mjs (hunk 3 above)
Comment (Skn0tt) [reply src/utils/static-server.mjs:24]: okay, but right now we're not setting `Server` anywhere, right? should we?

Comment (eduardoboucas) [reply src/utils/static-server.mjs:24]: We're not. Production gives you `server: Netlify` and CLI gives you `x-powered-by: netlify-dev`.

Comment (Skn0tt) [reply src/utils/static-server.mjs:24]: that's what I figured. This PR is about moving the CLI closer to prod together, so let's add `reply.header("Server", "Netlify")`

Comment (eduardoboucas) [reply src/utils/static-server.mjs:24]: I'm confused. That's what we're doing [here](https://github.com/netlify/cli/pull/5490/files#diff-5c5309633069037040a13b82d7abcfffe577c7516e6d356e46f407d729aa98a8R392), except we're doing it for all responses and not just the ones served by the static server.

Comment (Skn0tt) [reply src/utils/static-server.mjs:24]: I was confused. Didn't see that, sorry.

**Review by Skn0tt — APPROVED** [review]

**Pull request merged**